### DISCUSSION TITLE
Integrate opentelemetry agent jar with spring batch

### DISCUSSION THREAD

**Discussion by zhouzhou-myob** [open-telemetry/opentelemetry-java-instrumentation#9364] [Q&A] [answered]
Hi team, recently we are going to integrate spring batch with opentelemetry. I'm trying to use `opentelemetry-javaagent` to send our trace. But I found there is a related [discussion](https://github.com/open-telemetry/opentelemetry-java-instrumentation/discussions/2663) about this part. Seems like this is an "experimental" instrumentation. Not sure can we use this feature now

**Comment by mateuszrzeszutek** [comment]
Hey @zhouzhou-myob ,

You can use the instrumentation, you just have to pass an extra property `-Dotel.instrumentation.spring-batch.enabled=true`. We've marked it as an experimental instrumentation because at the time it was developed, we did not have a good idea about the shape of telemetry it was supposed to produce (well, we still don't have).

**Reply by zhouzhou-myob** [reply]
Much thanks @mateuszrzeszutek. There is another small thing want to confirm with you, in our spring batch job logic, we do some operations on AWS resources. Now we want to track all the batch logic and aws operations in the same trace. Not sure if we need any additional code change or dependency configurations during integration with the agent?

**Reply by mateuszrzeszutek** [reply, marked as answer]
> There is another small thing want to confirm with you, in our spring batch job logic, we do some operations on AWS resources. Now we want to track all the batch logic and aws operations in the same trace. Not sure if we need any additional code change or dependency configurations during integration with the agent?

The javaagent already includes AWS SDK instrumentation, most (if not all) things should work out of the box.

**Reply by doppleware** [reply]
Hi @mateuszrzeszutek I have enabled the system property you suggested but I'm only seeing multiple traces with jdbc spans and no spans for job start/end or tracing grouped in any way. Is that intentional?
